packages/dd-trace/src/config.js

```diff
@@ -44,6 +44,18 @@ function remapify (input, mappings) {
   return output
 }
 
+function propagationStyle (key, option, defaultValue) {
+  const envKey = `DD_TRACE_PROPAGATION_STYLE_${key.toUpperCase()}`
+  const envVar = process.env[envKey] || process.env.DD_TRACE_PROPAGATION_STYLE
+  const opt = typeof option === 'object' ? option[key] : option
+  const result = coalesce(opt, envVar, defaultValue)
+  if (Array.isArray(result)) return result
```

Comment:
Instead of having this check, let's just accept only an array for the option and only a string for the env var. This will keep everything simple and make types easier to document.

Reply:
That makes the handling more complicated though as we'd then have to do the string splitting stuff before we do the coalesce so we'd have to null-check multiple times. I'm also not a fan of the inconsistent input between sources. It makes more sense to only support the string form everywhere, but then I don't see a reason not to support the array form if it only take a single line to do so. 🤷🏻

Reply:
How does this look? Now only env vars will have the string splitting processing applied.